Sources/Shared/Extensions/URLExtensions.swift

```diff
@@ -114,6 +114,19 @@ private let ignoredSchemes = ["data"]
 private let supportedSchemes = permanentURISchemes.filter { !ignoredSchemes.contains($0) }
 
 extension URL {
+  /// Attempt to make a url that is sourced from a javascript `URL.href`.
+  ///
+  /// Since javascript's `URL` is more loose with acceptable caracters (such as `|`), we need to ensure such characters are escaped properly before we attempt to make a URL out of it.
+  public init?(javascriptURLString urlString: String) {
+    if let url = URL(string: urlString) {
+      self = url
+    } else if let escapedURLString = urlString.addingPercentEncoding(withAllowedCharacters: .URLAllowed), let url = URL(string: escapedURLString) {
+      self = url
+    } else {
+      return nil
+    }
+  }
+  
```

Comment:
Please move this (and its accompanying tests) to `BraveShared`/`BraveSharedTests`, we are not adding code to the Shared module anymore